frontend/src/components/common/Resource/MetadataDisplay.tsx

```diff
@@ -45,21 +45,35 @@ export function MetadataDisplay(props: MetadataDisplayProps) {
       return undefined;
     }
 
-    return ownerReferences.map((ownerRef, i) => (
-      <>
-        {ownerRef.kind in ResourceClasses ? (
-          <Link
-            routeName={new ResourceClasses[ownerRef.kind]().detailsRoute}
-            params={{ name: ownerRef.name, namespace: resource.metadata.namespace }}
-          >
-            {ownerRef.kind}: {ownerRef.name}
-          </Link>
-        ) : (
-          `${ownerRef.kind}: ${ownerRef.name}`
-        )}
-        {i < numItems - 1 && <br />}
-      </>
-    ));
+    return ownerReferences
+      .map((ownerRef, i) => {
+        if (ownerRef.kind in ResourceClasses) {
+          let routeName;
+          try {
+            routeName = new ResourceClasses[ownerRef.kind]().detailsRoute;
+          } catch (e) {
```

Comment:
An error shouldn't occur here, AFAIU. Or is there something I am missing?

Reply:
My guess is `ResourceClasses[ownerRef.kind]` isn't callable, so `new ResourceClasses[ownerRef.kind]()` fails.

Reply:
Okay. It should be callable as that's what we're setting those to be. But I am fine using it defensively here.